app/models/concerns/rule_remediation.rb

```diff
@@ -11,10 +11,14 @@ def remediation_issue_id
   private
 
   def short_profile_ref_id
-    profile = profiles.canonical.first
-    short_ref_id = profile.ref_id.downcase.split(
+    profile_ref_ids = profiles.collect(&:ref_id).to_set
```

Comment:
I guess my 2 cents here, :P favor SQL over Ruby:

```suggestion
    profile_ref_ids = profiles.distinct.pluck(:ref_id)
```

Reply:
The original implementation would not make an additional query, as profiles are prefetched in upper level(s). ;-)

Reply:
+1